sjsonnet/src/sjsonnet/PrettyYamlRenderer.scala

```diff
@@ -0,0 +1,434 @@
+package sjsonnet
+
+import java.io.{StringWriter, Writer}
+import java.util.regex.Pattern
+
+import ujson.BaseRenderer
+import upickle.core.{ArrVisitor, ObjVisitor}
+
+/**
+ * A version of YamlRenderer that tries its best to make the output YAML as
+ * pretty as possible: unquoted strings, de-dented lists, etc. Follows the PyYAML
+ * style. Also adds the ability to stream writes to a generic `Writer`.
+ */
+class PrettyYamlRenderer(out: Writer = new java.io.StringWriter(),
+                         indentArrayInObject: Boolean = false,
+                         indent: Int = 2,
+                         idealWidth: Int = 80) extends BaseRenderer[Writer](out, indent){
+  var newlineBuffered = false
+  var dashBuffered = false
+  var afterColon = false
+  var afterKey = false
+  var topLevel = true
+  var leftHandPrefixOffset = 0
+  var firstElementInArray = false
+  val newlinePattern = Pattern.compile("\n")
```

Comment:
There's already a copy of this in `YamlRenderer` and it is used below (along with other `YamlRenderer` methods)

Reply:
Good catch, it's dead